### PR TITLE
Standardize dividend argument in option models

### PR DESCRIPTION
**Pull request by alembcke**
This pull request fixes [#50](https://github.com/google/tf-quant-finance/issues/50).

## Motivation/Rational
The argument for dividends across different option pricing models used two names when only one was needed.  This pull request updates the files using `continuous_dividends` to instead use `dividend_rates`.  This makes it easier for end-users to reuse code when using the different option pricing models.

## Explanation of Changes
As requested in issue [#50](https://github.com/google/tf-quant-finance/issues/50), deprecation decorators were used to alert users to the change in argument names.  All argument names in the `tf_quant_finance/black_scholes/` directory and sub-directories for `continuous_dividends` were changed to `dividend_rates`.

## Testing
Many of the methods changed start with the variadic argument (*), which caused test to fail as Bazel would not recognize that the deprecated argument `continuous_dividends` was in the argument list, causing an error.  When the variadic argument was removed, tests completed successfully.

### CODE REVIEW

**Comment by alembcke** [comment]
@googlebot I signed it!

File: tf_quant_finance/black_scholes/approximations/american_option.py [hunk 1]
```diff
@@ -83,7 +86,7 @@ def adesi_whaley(*,
       where r are the discount rates, or risk free rates.
       Default value: `None`, equivalent to r = 0 and discount factors = 1 when
       discount_factors also not given.
-    continuous_dividends: An optional real `Tensor` of same dtype as the
+    dividend_rates: An optional real `Tensor` of same dtype as the
       `volatilities`. The continuous dividend rate on the underliers. May be
```
Comment (cyrilchim) [inline tf_quant_finance/black_scholes/approximations/american_option.py:90]: Could you please add `continuous_dividends ` to the docstring as well:

continuous_dividends:  `Tensor` equivalent to `dividend_rates`, to be deprecated. 

Here and elsewhere

File: tf_quant_finance/black_scholes/approximations/american_option.py [hunk 2]
```diff
@@ -20,8 +20,10 @@
 from tf_quant_finance.black_scholes import vanilla_prices
 from tf_quant_finance.math import gradient
 from tf_quant_finance.math.root_search import newton as root_finder_newton
+from tensorflow.python.util import deprecation
 
 
+@deprecation.deprecated_args(None, "continuous_dividends is deprecated. Use dividend_rates instead", "continuous_dividends")
 def adesi_whaley(*,
                  volatilities,
```
Comment (cyrilchim) [inline tf_quant_finance/black_scholes/approximations/american_option.py:28]: Thank you for cleaning up all methods. Could you please also modify Heston EU option pricer:
https://github.com/google/tf-quant-finance/blob/d99c50258cfcd12346f439d3bef1e733867cc707/tf_quant_finance/models/heston/approximations/european_option.py#L37

**Review by cyrilchim — COMMENTED** [review]
Thank you for the commit, Alex. Could you please also modify Heston option pricer?

Thank you!

File: tf_quant_finance/black_scholes/approximations/american_option.py (hunk 1 above)
Comment (alembcke) [reply tf_quant_finance/black_scholes/approximations/american_option.py:90]: Sure, it does seem to be a good idea to still reference `continuous_dividends` in the doc strings.

File: tf_quant_finance/black_scholes/approximations/american_option.py (hunk 2 above)
Comment (alembcke) [reply tf_quant_finance/black_scholes/approximations/american_option.py:28]: Sure, I updated the Heston EU option pricing model as well.  Also, there was one reference to `continuous_dividends` in the Local Volatility Model, where it called the Black Scholes option pricer, so I updated that as well.

**Comment by alembcke** [comment]
One more question, I also noticed that the API docs reference `continuous_dividends`, I assume that I should also update those references, do you agree?  If so, I will update those and push another commit.

**Comment by cyrilchim** [comment]
No worries about the API docs, they are synced via a script on a weekly basis

**Pull request merged**